defaults/main.yml

```
@@ -6,6 +6,8 @@
 ---
 gitlab_runner_version: ""
 
+gitlab_runner_pkg_version: "{{ gitlab_runner_version + '-1' if gitlab_runner_version | length == 0 or gitlab_runner_version is version('16.9.0', 'ge') else gitlab_runner_version }}"
```

I am not sure about the first condition, `gitlab_runner_version | length == 0`. Somehow I feel it should be `gitlab_runner_version | length > 0`. Am I wrong?

Should it be like this?


```suggestion
gitlab_runner_pkg_version: "{{ gitlab_runner_version + '-1' if gitlab_runner_version | length > 0 and gitlab_runner_version is version('16.9.0', 'ge') else gitlab_runner_version }}"
```

---

But if `gitlab_runner_version` string is empty, `gitlab_runner_pkg_version` should be empty as well and not `16.9.0`.

---

Currently, my interpretation would be that if `gitlab_runner_version = ""`, then the condition evaluates to true and as a result, the content of `gitlab_runner_pkg_version="-1"`. Am I wrong?

---

I see. Thanks, changed it.